### ISSUE TITLE
allow realtive offsets for line numbers

### ISSUE THREAD

**Issue by mProjectsCode** [mProjectsCode/obsidian-meta-bind-plugin#296]

**Comment by Tejeev** [comment]
Other options that have been discussed here and elsewhere that would be good to have are:
- prepend note
- append note
- targetted lines relative to the current line (perhaps 0 replaces the current line, -1 would be the line above the button, and 1 would be the one below?)
- specific numerical line in a note (exists)

**Comment by mProjectsCode** [comment]
Alright, so the current design plan would allow for very simple expressions (for line numbers in button actions) like the following:
- `selfStart`
- `selfEnd + 1`
- `fileStart-15`
- `fileEnd    +7`
- `4`

Simple numbers are still allowed, but on top of that simple addition and subtraction where the left-hand operand is a relative number and the right-hand operand is a user-specified number. Just a relative number should also be possible.

`selfStart` and `selfEnd` refer to the beginning of the button block (only for code block buttons) according to the markdown post processor section info. If the info is not available, the action should error when run.

`fileStart` and `fileEnd` refer to line `1` and the final line respectively.

Possibly `frontmatterStart` and `frontmatterEnd` for frontmatter offsets, defaulting to `fileStart` if not available (e.g. due to no frontmatter being present in the note).

**Comment by Tejeev** [comment]
@mProjectsCode Have you seen this? ^

**Comment by mProjectsCode** [comment]
Ah, seems like I overlooked this. Well-made PRs are always welcome.

**Comment by S1lentwanderer** [comment]
Any update on this? The Buttons Plugin seems to have broken sometime over the weekend, and I'm really missing the prepend functionality.

**Comment by jjmax75** [comment]
Hey folks, I'm not going to be able to do this now due to other commitments.

**Comment by Tejeev** [comment]
Thanks anyway @jjmax75.

**Comment by n1ck3** [comment]
Understandable. Thanks anyway, and thank you for a great plugin.

**Comment by Tejeev** [comment]
Yes, thanks @mProjectsCode for the great plugin and @jjmax75 for even the intent to offer a PR (which I can't even offer 😅).  I make heavy use of metabind and I appreciate the consideration and community that backs it up.

**Comment by mProjectsCode** [comment]
I released a test implementation of this FR in https://github.com/mProjectsCode/obsidian-meta-bind-plugin/releases/tag/1.3.4-canary.20250220T162034

**Comment by mProjectsCode** [comment]
This will be included in the next release.

**Comment by Tejeev** [comment]
@mProjectsCode are the `self` and `content`  relative to where the button is called?  I'm not sure I entirely understand the notes here.

**Comment by mProjectsCode** [comment]
There will be more detailed docs on this soon.

**Comment by Tejeev** [comment]
cool :)
just wanted to be sure I'd be able to make a button that would insert some template x lines relative to the actual button.

**Comment by PaperOrb** [comment]
> There will be more detailed docs on this soon.

Just checking to see if any docs were ever added for this?

**Comment by PaperOrb** [comment]
@mProjectsCode 

Just checking to see the status on the docs for this. Thanks!

**Comment by mProjectsCode** [comment]
Finally got around to writing the docs for this.
https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/buttons/#relative-line-numbers
